addons/mail/models/res_users.py

```diff
@@ -128,6 +161,70 @@ def unlink(self):
         self._unsubscribe_from_non_public_channels()
         return super().unlink()
 
+    def _notify_security_setting_update(self, subject, text, suggest_2fa=True, suggest_password_reset=True, **kwargs):
+        """ This method is meant to be called whenever a sensitive update is done on the user's account.
+        It will send an email to the concerned user warning him about this change and making some security suggestions.
+
+        This behavior can be manually disabled by turning off 'notify_security_updates' in the context
+
+        :param str subject: The subject of the sent email (e.g: 'Security Update: Password Changed')
+        :param str text: The text to embed within the email template (e.g: 'Your password has been changed')
+        :param bool suggest_2fa: Whether or not to suggest the end-user to turn on 2FA authentication in the email sent.
+          This parameter will only be taken into account when the 'auth_totp_mail' module is installed.
+          It is however placed here and within the initial xml template to avoid some complex overrides.
+        :param bool suggest_password_reset: Whether or not to suggest the end-user to reset his password in the email sent.
+          Defaults to True. """
+
+        if not self.env.context.get('notify_security_updates', True):
+            return
+
+        mail_create_values = []
+        for user in self:
+            body_html = self.env['ir.qweb']._render(
+                'mail.account_security_setting_update', {
+                    'company': user.company_id,
+                    'security_update_text': text,
+                    'suggest_2fa': suggest_2fa and user._notify_security_setting_update_suggest_2fa(),
+                    'suggest_password_reset': suggest_password_reset and self.env['ir.config_parameter'].sudo().get_param(
+                        "auth_signup.reset_password", True),
+                    'password_reset_url': f"{user.get_base_url()}/web/reset_password",
+                    'user': user,
+                },
+                minimal_qcontext=True,
+            )
+
+            body_html = self.env['mail.render.mixin']._render_encapsulate(
+                'mail.mail_notification_light',
+                body_html,
+                add_context={
+                    # the 'mail_notification_light' expects a mail.message 'message' context, let's give it one
+                    'message': self.env['mail.message'].sudo().new(dict(body=body_html, record_name=user.name)),
+                    'model_description': _('Account'),
+                    'company': user.company_id,
+                },
+            )
+
+            mail_create_values.append({
+                'auto_delete': True,
+                'author_id': user.partner_id.id,
+                'email_from': (
+                    user.company_id.partner_id.email_formatted or
+                    self.env.user.email_formatted or
```

Comment:
Note that if you updated your email, is the email really coming from your "new email" ? Normally company email takes precedence, so I suppose this is ok...